### DISCUSSION TITLE
Proposal: drop supporting for Go 1.15

### DISCUSSION THREAD

**Discussion by AsterDY** [bytedance/sonic#391] [Announcements]
Since new versions of Go have got significant improvements like `template generic`(Go 1.18), `register-based ABI` (Go 1.17) , we found that less and less online applications are still using Go 1.15. On the other hand, keeping backward compatible with old versions of Go costs a lot of our energy, especially for developing version-related packages like `sonic/loader`, `coder/assembler`. Thus, we want to drop supporting for Go 1.15 in our next middle version -- **v1.x.0**. In my perspective of view, it won't hurt much to upgrade your application from Go 1.15 to higher version.

**Comment by iwanlebron** [comment]
Support!!!

**Reply by Elli2111** [reply]
Ja keine Ahnung

**Comment by Clownsw** [comment]
支持
